central/group/datastore/validate.go

```diff
@@ -0,0 +1,54 @@
+package datastore
+
+import (
+	"github.com/gogo/protobuf/proto"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/uuid"
+)
+
+// groupIDPrefix should be prepended to every human-hostile ID of a
+// signature integration for readability, e.g.,
+//     "io.stackrox.group.94ac7bfe-f9b2-402e-b4f2-bfda480e1a13".
+const groupIDPrefix = "io.stackrox.group."
+
+// ValidateGroup validates the given group for conformity.
+// A group must fulfill the following:
+//	- have valid properties (validated via ValidateProps).
+//	- have a role name set.
+func ValidateGroup(group *storage.Group) error {
+	if group.GetProps() == nil {
+		return errox.InvalidArgs.New("group properties must be set")
+	}
+	if err := ValidateProps(group.GetProps()); err != nil {
```

Comment:
This might be problematic when `ValidateGroup` is called in e.g. `Update(), Upsert(), Mutate()`

Reply:
I've removed the ID check within `ValidateGroup`, as we currently do not want to more lax, supporting the previous way of retrieving groups via their properties. I've added a comment to later on add the ID within the validation.